runtime-sdk/modules/contracts/src/lib.rs

```diff
@@ -475,6 +475,12 @@ impl<Cfg: Config> Module<Cfg> {
             params: &params,
         };
         let result = wasm::instantiate::<Cfg, C>(&mut exec_ctx, &contract, &body);
+
+        // Always return success in CheckTx, as we might not have up-to-date state.
```

Comment:
This will skip estimating gas for processing the result?

Reply:
Hm so this should be something like:

```rust
        let result = match results::process_execution_result(ctx, result) {
            Ok(ok) => ok,
            Err(e) => {
                // Always return success in CheckTx, as we might not have up-to-date state.
                if ctx.is_check_only() {
                    return Ok(types::InstantiateResult::default());
                } else {
                    return Err(e);
                }
            }
        };
        if let Err(e) =
            results::process_execution_success::<Cfg, C>(ctx, &params, &contract, result)
        {
            // Always return success in CheckTx, as we might not have up-to-date state.
            if !ctx.is_check_only() {
                return Err(e);
            }
        }
```

right?

Reply:
Yeah this looks quite ugly, could maybe make it nicer it with a closure that returns a result?

Reply:
Yeah this looks quite ugly, could maybe make it nicer it with a closure that returns a result?

Reply:
Yeah agreed, pushed a version with a closure.

Reply:
Nice!